opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -1959,10 +1962,175 @@ std::vector<double> EclipseGrid::createDVector(const std::array<int,3>& dims, st
+    }
+
+    std::size_t EclipseGrid::activeIndexLGR(const std::string& label, std::size_t localIndex) const {
+        if (lgr_label.compare(label) == 0 ){
+            std::size_t local_global_ind = getActiveIndex(localIndex);
+            this->assertIndexLGR(local_global_ind);
+            return lgr_level_active_map[local_global_ind] + lgr_global_counter;
+        }
+        else if (lgr_children_cells.size() == 0) {
+            return 0;
+        }
+        else {
+            std::size_t local_global_ind = 0;
+            for (const auto& lgr_cell : lgr_children_cells) {
+                local_global_ind += lgr_cell.activeIndexLGR(label, localIndex);
+            }
+            return local_global_ind;
+        }   
+    }
+
+
+    std::size_t EclipseGrid::activeIndexLGR(const std::string& label, std::size_t i, std::size_t j, std::size_t k) const {
+        if (lgr_label.compare(label) == 0 ){
+            this->assertIJK(i,j,k);
+            std::size_t local_global_ind = getActiveIndex(i,j,k);
+            this->assertIndexLGR(local_global_ind);
+            return lgr_level_active_map[local_global_ind] + lgr_global_counter;
+        }
+        else if (lgr_children_cells.size() == 0){
+            return 0;
+        }
+        else {
+            std::size_t local_global_ind = 0;
+            for (const auto& lgr_cell : lgr_children_cells) {
+                // better add the bases a priori
+                local_global_ind += lgr_cell.activeIndexLGR(label, i,j,k);
+            }
+            return local_global_ind;
+        }   
+    }
+    
+    std::size_t EclipseGrid::getTotalActiveLGR() const
+    {
+        std::size_t num_coarse_level =  getNumActive();
+        std::size_t num_fine_cells = 0;
+        std::size_t num_spanned_cells = 0;
+        for (const auto&  fine_cells : lgr_children_cells) {
+            num_fine_cells += fine_cells.getTotalActiveLGR();
+            num_spanned_cells += fine_cells.get_father_global().size();
+
+        }
+        return num_coarse_level + num_fine_cells - num_spanned_cells;
+    }  
+
+
+    void EclipseGrid::init_lgr_cells(const LgrCollection& lgr_input) {
+        save_all_lgr_labels(lgr_input);
+        create_lgr_cells_tree(lgr_input);
+        init_lgr_cells_index();
+        init_lgr_global_cells_index();
+    }
+    
+    void EclipseGrid::save_all_lgr_labels(const LgrCollection& lgr_input) {
+        all_lgr_labels.reserve(lgr_input.size()+1);
+        all_lgr_labels.push_back("GLOBAL");
+        for (std::size_t index = 0; index < lgr_input.size(); index++) {
+            all_lgr_labels.push_back(lgr_input.getLgr(index).NAME());
+        }    
+    }
+
+    void EclipseGrid::create_lgr_cells_tree(const LgrCollection& lgr_input) {                
+          auto IJK_global = [this](const auto& i_list, const auto& j_list, const auto& k_list){
+            if ((i_list.size() != j_list.size()) || (j_list.size() != k_list.size()) || (k_list.size() != i_list.size()) ){
```

Comment:
You don't need to check all three of these conditions.  If `i.size() == j.size() && j.size() == k.size()` then `i.size() == k.size()` by transitivity.